### PR TITLE
typing: update options parameter type in `ui.ActionRow.add_select`

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Previous annotations didn't allow calls like `row.add_select(options=["a", "b"])`, full `SelectOption` instances had to be provided even though the underlying code supports both.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
This needs a documentation change for `disnake.ui.ActionRow.add_select`, should use the same annotation in the docstring as `disnake.ui.Select`.

**Comment by shiftinv** [comment]
> This needs a documentation change for `disnake.ui.ActionRow.add_select`, should use the same annotation in the docstring as `disnake.ui.Select`.

https://github.com/DisnakeDev/disnake/pull/744/commits/0379f4dd59b97c7bc54aa7fd3afb864b496e8fc9

**Comment by onerandomusername** [comment]
I wonder if it makes sense to make a replacement string for the annotation, as it's a complex annotation used in multiple locations

File: changelog/744.misc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Update annotation and description of ``options`` parameter of :func:`ui.ActionRow.add_select <disnake.ui.ActionRow.add_select>` to match :class:`ui.Select <disnake.ui.Select>`.
```
Comment (onerandomusername) [inline changelog/744.misc.rst:1]: This is a docs change, not a miscellaneous change, is it not?

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: changelog/744.misc.rst (hunk 1 above)
Comment (shiftinv) [reply changelog/744.misc.rst:1]: It's kind of both a typing change and a docs change, but adding two changelog fragments for something minor like this felt a bit excessive.

**Comment by shiftinv** [comment]
> I wonder if it makes sense to make a replacement string for the annotation, as it's a complex annotation used in multiple locations

Good point, however it only appears three times, don't think that's worth it.

**Pull request merged**